### DISCUSSION TITLE
DB2 IBM view system names

### DISCUSSION THREAD

**Discussion by Starcomfr** [dbeaver/dbeaver#16082] [Ideas]
Hi,

It would be usefull to be able to see system names for tables and columns on DB2 IBM.

Sytem table names can be knownusing request like:
`SELECT SYS_TNAME FROM qsys2.systables WHERE TABLE_SCHEMA = 'My_schema' AND TABLE_NAME = 'My_table_name';`

System column names can be knownusing request like:
`SELECT SYSTEM_COLUMN_NAME FROM sysibm.sqlcolumns WHERE table_schem = 'My_Schema' AND TABLE_NAME = 'My_table_name' ORDER BY ORDINAL_POSITION;`

**Comment by LonwoLonwo** [comment]
Hello @Starcomfr 

What issue with the current view system names do you have?

**Reply by Starcomfr** [reply]
Hello @LonwoLonwo ,

It's not an really an issue but an idea.

![dbeaver_2022-04-06_15-18-58](https://user-images.githubusercontent.com/8850081/161988871-b8f75482-7f5d-4a01-9f2d-d48051b28a6a.png)
As you can see on the screenshot, actually it's only possible to display column names in the view of the table properties.
Therefore a short name exists for column on DB2 IBM witch is created using "FOR COLUMN".

ie:
`CREATION_DATE FOR COLUMN CREADET TIMESTAMP NOT NULL DEFAULT CURRENT TIMESTAMP`

A short system name for the table is not visible in the table properties using DBeaver 
![dbeaver_2022-04-06_15-20-43](https://user-images.githubusercontent.com/8850081/161989010-d5b92a07-4ec5-44a5-a275-75f0a79cd4fa.png)

This name is created using 'FOR SYSTEM NAME' as:

`CREATE TABLE MY_LONG_TABLE_NAME FOR SYSTEM NAME MYTABNAME (`

These informations can be found on IBM system using the request specified in my first message.

**Reply by LonwoLonwo** [reply]
Ok, maybe we can add this. You can create a ticket: https://github.com/dbeaver/dbeaver/issues/new/choose

**Reply by Starcomfr** [reply]
Thanks a lot !

Done: Ticket #16102
